### PR TITLE
 Testing

### PR DESCRIPTION
**Pull request by AbdallahAmmar96**
- Install tap and tap-spec
- make two folder one for endpoint and the other for queries
- and write and run an initial test 
### For index.js
The function is read the folders name and then read every test file in each folder and run it 

Relates #11

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +23,8 @@
   },
   "homepage": "https://github.com/FACG6/AMiDEAST#readme",
   "devDependencies": {
-    "concurrently": "^4.1.0"
+    "concurrently": "^4.1.0",
+    "tap-spec": "^5.0.0",
```
Comment (kannanhassouna) [inline package.json]: move these modules into ``` server/package.json ```

File: package.json [hunk 2]
```diff
@@ -5,8 +5,7 @@
   "main": "index.js",
   "scripts": {
     "dev": "NODE_ENV=development nodemon server/index.js",
-    "test:query": "NODE_ENV=testing tape ./server/test/query-test.js | tap-spec",
-    "test:router": "NODE_ENV=testing tape ./server/test/router-test.js | tap-spec",
+    "test": "node ./server/test | tap-spec",
```
Comment (kannanhassouna) [inline package.json]: and here move it to ``` server/package.json ```

File: server/test/index.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const testFolder = './server/test';
```
Comment (kannanhassouna) [inline server/test/index.js]: i think you don't need to all of these lines.
if you just require the files here that will be fine 
like : 
```
require("./Endpoints/router-test");
require("./Queries/query-test");
```

File: server/test/Queries/query-test.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const tape = require('tape');
+tape('just for testing', (t) => {
```
Comment (kannanhassouna) [inline server/test/Queries/query-test.js:3]: white line

File: server/test/Endpoints/router-test.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const tape = require('tape');
+tape('just for testing', (t) => {
```
Comment (kannanhassouna) [inline server/test/Endpoints/router-test.js:3]: also here

**Review by kannanhassouna — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (kannanhassouna) [reply package.json]:
```suggestion
    "test": "NODE_ENV=testing node ./server/test | tap-spec",
```

**Review by kannanhassouna — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (AbdallahAmmar96) [reply package.json]: Done

File: server/test/index.js (hunk 3 above)
Comment (AbdallahAmmar96) [reply server/test/index.js]: ok bsh mohndes @knanahassouna1

File: package.json (hunk 1 above)
Comment (AbdallahAmmar96) [reply package.json]: No, I make it on purpose,
where is the problem in that

Comment (kannanhassouna) [reply package.json]: because you will not use it on the client side, just on the server side , not in both

**Review by kannanhassouna — CHANGES_REQUESTED** [review]

**Review by kannanhassouna — APPROVED** [review]

**Pull request merged**